editor/plugins/game_view_plugin.cpp

```diff
@@ -329,8 +505,100 @@ Dictionary GameView::get_state() const {
 	return d;
 }
 
-GameView::GameView(Ref<GameViewDebugger> p_debugger) {
+void GameView::set_window_layout(Ref<ConfigFile> p_layout) {
+	floating_window_rect = p_layout->get_value("GameView", "floating_window_rect", Rect2i());
+	floating_window_screen = p_layout->get_value("GameView", "floating_window_screen", -1);
+	floating_window_screen_rect = p_layout->get_value("GameView", "floating_window_screen_rect", Rect2i());
+}
+
+void GameView::get_window_layout(Ref<ConfigFile> p_layout) {
+	if (window_wrapper->get_window_enabled()) {
+		_update_floating_window_settings();
+	}
+
+	p_layout->set_value("GameView", "floating_window_rect", floating_window_rect);
+	p_layout->set_value("GameView", "floating_window_screen", floating_window_screen);
+	p_layout->set_value("GameView", "floating_window_screen_rect", floating_window_screen_rect);
+}
+
+void GameView::_update_floating_window_settings() {
+	if (window_wrapper->get_window_enabled()) {
+		floating_window_rect = window_wrapper->get_window_rect();
+		floating_window_screen = window_wrapper->get_window_screen();
+		floating_window_screen_rect = DisplayServer::get_singleton()->screen_get_usable_rect(floating_window_screen);
+	}
+}
+
+void GameView::_update_arguments_for_instance(int p_idx, List<String> &r_arguments) {
+	if (p_idx != 0 || !embed_on_play || !DisplayServer::get_singleton()->has_feature(DisplayServer::FEATURE_WINDOW_EMBEDDING)) {
+		return;
+	}
+
+	// Remove duplicates/unwanted parameters.
+	List<String>::Element *E = r_arguments.find("--position");
```

Comment:
This is a linked list, so finding/erasing repeatedly is very costly. You can iterate the list and remove arguments as you find them.
```C++
List<String>::Element *E = r_arguments.front();
while (E) {
	N = E->next();

	if (E->get() == "something") {
		r_arguments.erase(E);
	} else if (E->get() == "something else") {
		r_arguments.erase(E);
	}

	E = N;
}
```
etc.

Reply:
Done!